### ISSUE TITLE
Auditing not working when using FeatureStoreHttp

### ISSUE THREAD

**Issue by abuinteam** [clun/ff4j#267]
Hi,

In my webapplication (client) I am initializing the ff4j as HttpStore using restAPI like below

ff4j = new FF4j();           
ff4j.setFeatureStore(new FeatureStoreHttp("http://localhost:9081/api/ff4j","FF4J","XXX"));            ff4j.setPropertiesStore(new PropertyStoreHttp("http://localhost:9081/api/ff4j","FF4J","XXX"));

In my admin console application/rest api springbootstarter project I am using the below config,

      FF4j ff4j = new FF4j();
      ff4j.setFeatureStore(new FeatureStoreSpringJdbc(dataSource));
        ff4j.setPropertiesStore(new PropertyStoreSpringJdbc(dataSource));
        ff4j.setEventRepository(new EventRepositorySpringJdbc(dataSource));
        ff4j.autoCreate(true);
        ff4j.audit(true);
       
      Whenever I hit my application URL it is invoking the API and getting the feature result whether it is enabled or not. But if I go to my admin console I don't see any audit details for that feature.

But if I directly access those features inside admin console app, audit gets inserted/updated.

Any help would be appreciated.

**Comment by clun** [comment]
Pretty busy this week. I will into into ALL your findings pretty soon.

**Comment by abuinteam** [comment]
Thank you very much

**Comment by clun** [comment]
In the Springboot web api the featureStore is autowired directly and do not use the `FF4j` bean.

https://github.com/clun/ff4j/blob/master/ff4j-spring-boot-web-api/src/main/java/org/ff4j/spring/boot/web/api/resources/FeatureStoreResource.java#L56-L58

Will be part of same commit as security.

**Comment by paul58914080** [comment]
I think the problem is you have configured the audit trail on admin console and not in your client ?

**Comment by clun** [comment]
I think we should setup audit in the Server that expose the API (maybe the admin console as you mentioned). Not in all client because they may not have access to the underlying storage right ?

Today if we do that, the FeatureStoreResource does not use the FeatureStoreAuditProxy as it autowired the FeatureStoreItself. My idea was to inject FF4J and access the featureStore with ff4j.getFeatureStore() in order to use the audit part.

**Comment by paul58914080** [comment]
Finally after a bit of investigation I was able to find the issue

https://github.com/clun/ff4j/blob/master/ff4j-web/src/main/java/org/ff4j/web/embedded/ConsoleServlet.java#L261

is using the instance of FF4j client. 

Essentially it should be 

`getFf4j().getFeatureStore()`

therefore where you turn on or turn off the feature through the admin console of the client it does not reflect.

This works perfectly fine with PropoertyStore because getFf4j().getFeatureStore()

https://github.com/clun/ff4j/blob/master/ff4j-web/src/main/java/org/ff4j/web/embedded/ConsoleServlet.java#L218

@clun let me know if you agree with me. I will fix this

**Comment by abuinteam** [comment]
Thanks. I initialized the audit in the server only not on the client.

**Comment by paul58914080** [comment]
You dont have to . I think I fixed it with 4aa175622206255f8c7ebf75925e04a211186b22

**Comment by abuinteam** [comment]
ok thanks. where can I download the latest fixes?

**Comment by paul58914080** [comment]
I have just committed the code. It is yet to be released. I would let @clun comment when he plans to release :)

**Comment by clun** [comment]
I still have to fix the security issue in spring-boot-web-api (ApiConfig with authentication enabled) and we will be able to release. I have plenty of time tomorrow should be ok !!

**Comment by clun** [comment]
Part of 1.7 release

**Comment by abuinteam** [comment]
Hello Clun,

I didn't get time to verify this last week. I was trying to validate 1.7 changes. After I have download the 1.7 related jars and dependencies I was getting an run time exception and I could not able to pass that after a lot of options. 

When I was referring ff4j tab library in my jsp and I was accessing my jsp page I am getting the below exception.

Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.swagger.jaxrs.json.JacksonJsonProvider
	at org.ff4j.web.api.security.ClientHttpJersey1Utils.buildJersey1Client(ClientHttpJersey1Utils.java:54)
	at org.ff4j.web.jersey1.store.FeatureStoreHttp.getJerseyClient(FeatureStoreHttp.java:146)
	at org.ff4j.web.jersey1.store.FeatureStoreHttp.getStore(FeatureStoreHttp.java:159)
	at org.ff4j.web.jersey1.store.FeatureStoreHttp.exist(FeatureStoreHttp.java:226)
	at org.ff4j.audit.proxy.FeatureStoreAuditProxy.exist(FeatureStoreAuditProxy.java:207)
	at org.ff4j.FF4j.exist(FF4j.java:414)
	at org.ff4j.web.taglib.AbstractFeatureTag.doStartTag(AbstractFeatureTag.java:84)
	at com.ibm._jsp._menuTree._jspService(_menuTree.java:242)
	... 20 more


The below are my jars on the class path,

![2017-12-10 23_49_51-eclipse-workspace - teamonlinesystem_webroot_jsp_include jsp - eclipse](https://user-images.githubusercontent.com/33212676/33820588-05a5b974-de05-11e7-926c-e64f52c670f1.png)

Am I missing something?

**Comment by abuinteam** [comment]
I passed this error by resolving the dependencies. And below are the observations.

1. **checkOn** event is not getting inserted when I access the feature from JSP or Java Class. Was working fine 1.6.5. Since the event was not loading graphs are not working.

2. In the ff4j_audit table, EVENT_NAME accepts maximum of 30 characters. Not holding big names. I changed the SQL column width and this is working fine. Is there anyway where I can get the latest SQL DDL changes?
